cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringHandler.java

```diff
@@ -101,10 +112,24 @@ public void getTether(HttpRequest request, HttpResponder responder, @PathParam("
   @DELETE
   @Path("/tethering/connections/{peer}")
   public void deleteTether(HttpRequest request, HttpResponder responder, @PathParam("peer") String peer)
-    throws PeerNotFoundException, IOException {
+    throws NotFoundException, IOException, BadRequestException {
+    boolean tetheringServerEnabled = cConf.getBoolean(Constants.Tethering.TETHERING_SERVER_ENABLED);
+    if (tetheringServerEnabled) {
+      // Fail deletion if there are any profiles using this tethering
+      List<Profile> tetheringProfiles = profileService.getProfiles(
```

Comment:
If we allow deletion when the profile is disabled, then the profile can be re-enabled => we'll be back the same situation that we're trying to prevent, i.e. active profile is using a tethering connection that doesn't exist